lib/packaging/paths.rb

```
@@ -31,9 +31,23 @@ def tag_from_artifact_path(path)
     # if we didn't find a platform or a version, probably a codename
     if platform.nil? || version.nil?
       codename = Pkg::Platforms.codenames('deb').find { |c| path =~ /\/#{c}/ }
-      fail "I can't find a codename or platform in #{path}, teach me?" if codename.nil?
+      if codename.nil?
+        if fail_hard
+          fail "I can't find a codename or platform in #{path}, teach me?"
+        else
```

Any reason for not always failing hard?

---

Same as below. It's a poorly written method. I had updated this to handle this warn case with more grace, but that involved quiet a bit of change. I felt uncomfortable including all of that in this PR. I opened a new ticket to deal with that.